### ISSUE TITLE
Stories overview in sidebar

### ISSUE THREAD

**Issue by androdel** [storybookjs/storybook#6588]
Hi,

I have 2 different stories, but I only go to the specific path of the last story I edited.
Initially, in the url I see story/*, but then redirects. I used the Angular docs to setup the config to load all stories. You would expect that this works out of the box.

Latest versions Angular and Storybook are being used.

**Comment by shilman** [comment]
Can you rephrase your question or elaborate, maybe include a screenshot? I don't understand your question

**Comment by androdel** [comment]
I have 2 stories:
<img width="224" alt="Schermafbeelding 2019-04-23 om 21 30 46" src="https://user-images.githubusercontent.com/13794129/56610144-18e95c00-660f-11e9-95a1-82ca20bdb0c5.png">

I load them with the following code:
`import {configure, addDecorator, moduleMetadata} from "@storybook/angular";
import {I18nModule} from "../src/app/i18n.module";

addDecorator(
  moduleMetadata({
    imports: [I18nModule],
  }),
);

// automatically import all files ending in *.stories.ts
const req = require.context("../src/stories", true, /\.stories\.ts$/);
function loadStories() {
  req.keys().forEach(filename => req(filename));
}

configure(loadStories, module);
`
When I go to my localhost where I serve my Storybook, I see this:
<img width="1634" alt="Schermafbeelding 2019-04-23 om 21 36 53" src="https://user-images.githubusercontent.com/13794129/56610547-f86dd180-660f-11e9-93e4-7a924052006c.png">

The url is the following: http://localhost:6006/?path=/story/header-account--default

As you can see in the sidebar, there is only on story/component visible.
If I make a change to my other story, then that one is shown.

I've read that HMR could be a possible cause of this issue.

**Comment by shilman** [comment]
Weird... 🤷‍♂️ Seems kind of tricky. Has anybody else seen this? Is there a repro somewhere?

**Comment by kroeder** [comment]
Yes, but only during hmr. This bug exists since 3.x or so (maybe longer?). It works fine in prod and is only an issue during dev (at least for me)

The exact behavior in my case 
- run start-storybook: everything is fine 
- do a change in a story: the changed story is still there but a handful of others disappear in the navigation bar 

This can be fixed by reloading the page (F5) 

But it sounds like @androdel initially does not see every story? 

@shilman you might be able to reproduce this in our example app. Just follow my steps and do a couple of changes for hmr

**Comment by androdel** [comment]
Would disabling HMR be a workaround?

**Comment by kroeder** [comment]
Without HMR you'd need to reload the whole page after every code change

**Comment by androdel** [comment]
@kroeder That's fine. Not having the whole list of stories is more of a pain for me. Is there a way that I can enforce this or do I have to do it in the code?
